### ISSUE TITLE
Regression: iTop 3.1: hits memory limits

### ISSUE THREAD

**Issue by jbostoen** [jbostoen/itop-jb-report-generator#10]
Seems to do with changed behavior in iTop 3.1.0 regarding REST.

The memory peak is in the conversion of an object set to JSON.

**Comment by jbostoen** [comment]
After analysis, the problem is easily reproducible.
First, create 5,000 tickets.

Then, have some code like

```
$sStart = date('H:i:s');
$oFilter = DBObjectSearch::FromOQL('SELECT UserRequest');
$oSet = new DBObjectSet($oFilter);
while($o = $oSet->Fetch()) {
			
}
		
die('Start: '.$sStart.' - End: '.date('H:i:s').' - Memory peak: '.memory_get_peak_usage(true));

```

Surpisingly: for just one object, iTop 3.1 uses less memory. For a loop, it uses a whole lot more.

**Comment by jbostoen** [comment]
Fixed as of iTop 3.1.0-3.
